processor/eqreceipt/app.go

```diff
@@ -0,0 +1,124 @@
+package eqreceipt
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/ONSdigital/census-rm-pubsub-adapter/models/incoming-pubsub"
+	"github.com/ONSdigital/census-rm-pubsub-adapter/models/rabbit"
+	"log"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/pkg/errors"
+	"github.com/streadway/amqp"
+)
+
+type App struct {
+	RabbitConn   *amqp.Connection
+	RabbitChan   *amqp.Channel
+	PubSubClient *pubsub.Client
+	EqRecieptSub *pubsub.Subscription
+	ReceiptChan  chan incoming_pubsub.EqReceipt
+}
+
+func (a *App) Setup(ctx context.Context, rabbitConnectionString, projectID string) {
```

Comment:
You could phrase this as 'New()' which returns an App pointer. It saves you creating the empty struct in main just to call this func.

Reply:
At this point, the package name comes into play here, because creating an app and then setting it up is clear, but `eqReceipt.New()` might sound a bit more like you're creating a data struct rather than a service. That might be another indication this should all be at the 'processor' package level. `processor.New()` feels pretty accurate for what's happening here